File: BlogApp-RN/src/features/posts/domain/usecases/addComment.ts
import { AsyncResult } from "../../../../core/typedefs/asyncResult";
import { CompletedComment } from "../../data/models/completedComment";
import PostRepo from "../repos/postRepo";

class AddComment {
  constructor(private repo: PostRepo) {}
  call: (
    postId: string,
    comment: string,
    author: string
  ) => AsyncResult<CompletedComment> = (postId, comment, author) => {
    return this.repo.addComment(postId, comment, author);
  };
}

export default AddComment;

File: blogapp_flutter/lib/features/posts/domain/usecases/add_comment.dart
import 'package:blogapp_flutter/core/typedefs/async_result.dart';
import 'package:blogapp_flutter/core/typedefs/usecases.dart';
import 'package:blogapp_flutter/features/posts/data/models/completed_comment.dart';
import 'package:blogapp_flutter/features/posts/domain/repos/posts_repo.dart';

class AddComment extends UsecaseWithParams<CompletedComment, AddCommentParams> {
  final PostsRepo _repo;
  AddComment(this._repo);
  @override
  AsyncResult<CompletedComment> call(params) {
    return _repo.addComment(params.postId, params.comment, params.author);
  }
}

class AddCommentParams {
  final String postId;
  final String comment;
  final String author;
  AddCommentParams(
      {required this.postId, required this.comment, required this.author});
}
